File: app/src/main/java/com/fallgamlet/dnestrcinema/utils/HttpUtils.java
package com.fallgamlet.dnestrcinema.utils;

import android.text.Html;
import android.text.Spanned;

/**
 * Created by fallgamlet on 22.03.17.
 */

public abstract class HttpUtils {
    public static String getAbsoluteUrl(String baseUrl, String url) {
        if (url == null) {
            return null;
        }

        url = url.trim();
        if (url.toLowerCase().startsWith("http://")) {
            return url;
        }

        if (baseUrl == null) {
            return null;
        }

        baseUrl = baseUrl.trim();

        String dopSlash = url.startsWith("/") ? "": "/";
        if (baseUrl.endsWith("/")) {
            baseUrl = baseUrl.substring(0, baseUrl.length()-1);
        }
        return baseUrl+dopSlash+url;
    }

    public static Spanned fromHtml(String html) {
        if (html == null) {
            return null;
        }
        Spanned result;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
            result = Html.fromHtml(html,Html.FROM_HTML_MODE_LEGACY);
        } else {
            result = Html.fromHtml(html);
        }
        return result;
    }
}

File: app/src/main/java/com/fallgamlet/dnestrcinema/utils/HttpUtils.kt
package com.fallgamlet.dnestrcinema.utils

import android.text.Html
import android.text.Spanned


object HttpUtils {
    fun getAbsoluteUrl(baseUrl: String?, url: String?): String? {
        url ?: return null

        var prefixUrl = baseUrl
        var postfixUrl: String? = url

        postfixUrl = postfixUrl!!.trim { it <= ' ' }
        if (postfixUrl.toLowerCase().startsWith("http://")) {
            return postfixUrl
        }

        prefixUrl ?: return null
        prefixUrl = prefixUrl.trim(' ')

        val dopSlash = if (postfixUrl.startsWith("/")) "" else "/"
        if (prefixUrl.endsWith("/")) {
            prefixUrl = prefixUrl.substring(0, prefixUrl.length - 1)
        }

        return prefixUrl + dopSlash + postfixUrl
    }

    fun fromHtml(html: String?): Spanned? {
        html?: return null

        return if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
            Html.fromHtml(html, Html.FROM_HTML_MODE_LEGACY)
        } else {
            Html.fromHtml(html)
        }
    }
}
